services/src/main/scala/cromwell/services/instrumentation/impl/statsd/StatsDConfig.scala

```diff
@@ -0,0 +1,35 @@
+package cromwell.services.instrumentation.impl.statsd
+
+import cats.data.Validated._
+import cats.syntax.apply._
+import cats.syntax.validated._
+import com.typesafe.config.Config
+import lenthall.exception.MessageAggregation
+import lenthall.validation.ErrorOr.ErrorOr
+import lenthall.validation.Validation._
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.duration.FiniteDuration
+
+case class StatsDConfig(hostname: String, port: Int, prefix: Option[String], flushRate: FiniteDuration)
+
+object StatsDConfig {
+  def apply(serviceConfig: Config): StatsDConfig = {
+    val statsDConfig = serviceConfig.getConfig("statsd")
+
+    val hostname: ErrorOr[String] = validate[String] { statsDConfig.as[String]("hostname") }
+    val port: ErrorOr[Int] = validate[Int] { statsDConfig.as[Int]("port") }
+    val prefix: ErrorOr[Option[String]] = statsDConfig.as[Option[String]]("prefix").validNel
+    val flushRate: ErrorOr[FiniteDuration] = validate[FiniteDuration] { statsDConfig.as[FiniteDuration]("flush-rate") }
+
+    (hostname, port, prefix, flushRate) mapN { (h, p, n, f) => 
+      new StatsDConfig(h, p, n, f)
```

Comment:
ToL I think you can do `mapN(StatsDConfig.apply.tupled)`

Reply:
I couldn't get this to work, I get an `ambiguous reference to overloaded definition` because there are 2 `apply` methods. I'm sure there's a way to get it to use the right one but it didn't seem like it was worth fighting with this. If you know how to do it though I'd be curious.